tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/KnowledgeGuardian.scala

```diff
@@ -73,6 +78,52 @@ private[emitter] final class KnowledgeGuardian {
     invalidateAll
   }
 
+  private def computeHasInlinedInit(linkingUnit: LinkingUnit): Set[String] = {
+    /* Those classes are instantiated in scalajsenv.js. Since they have
+     * multiple constructors and/or are not final, scalajsenv.js is written
+     * with the assumption that they will not have an inlined init. We
+     * therefore blacklist them here so that this is always true.
+     */
+    val blackList = Set(
+        "jl_ClassCastException",
+        "jl_ArrayIndexOutOfBoundsException",
+        "sjsr_UndefinedBehaviorError",
+        "js_CloneNotSupportedException"
+    )
+
+    val scalaClassDefs = linkingUnit.classDefs.filter(_.kind.isClass)
+    val instantiatedClasses = scalaClassDefs.filter(_.hasInstances)
```

Comment:
Consider inlining this below to make it explicit that it is unused later. Being able to give it a name does not add any value IMO.